### PR TITLE
(WIP) Add getOptionValue

### PR DESCRIPTION
**Pull request by jasonLaster**
Based on our discussion, here's a basic implementation for a helper getOptionValue which composes two of our most used helpers.

I'll add docs and tests if we like it

### CODE REVIEW

**Comment by paulfalgout** [comment]
Also I suppose there's a reason getValue can't handle a string as an option?

**Comment by jasonLaster** [comment]
So, I'm not convinced that there are a lot of uses, the motivation is that our DSL, which evaluates properties like `modelEvents` or `behaviors` for instance properties or options properties, object properties or function properties, is prone for error. Here's a [gist](https://gist.github.com/jasonLaster/fd83341e87665ea41a59) of the places where `getOption` is called. Unfortunately, we can't assume that every `getOption` call intends to immediately evaluate a function


##### Types:
1. (instance, object)  `View.extend({modelEvents: {}});`
2. (instance, fnc)  `View.extend({modelEvents: function() {return {}}});`
3. (options, object) `new View({modelEvents: {}})`
4. (options, fnc)   `new View({modelEvents: function() {return {}}})`

**Comment by paulfalgout** [comment]
> Unfortunately, we can't assume that every getOption call intends to immediately evaluate a function

My 2nd questions is essentially the opposite.. Can we assume that all `getValue`s could handle a string as a `getOption`?

**Comment by paulfalgout** [comment]
FWIW I think this pattern happens a couple other times but in separate steps.. For instance I ran into it kind of on the radio mixin
`getOption` here https://github.com/marionettejs/backbone.marionette/pull/2758/files#diff-951bb4cc49ca2d8446383594f46fe145R26
passed to something that `getValue`s https://github.com/marionettejs/backbone.marionette/pull/2758/files#diff-73c6cec277d5cc65ad42a85db80263adR31

**Comment by paulfalgout** [comment]
Hmm I don't think `getValue` can take a string and assume `getOption`.. sometimes the field we're analyzing is either a string or a function returning a string, and we want the string to just pass through... so yeah.. :+1: Just needs tests, docs I guess.. though we have a general v3 doc issue.. and any other implementations in the src.

**Comment by jasonLaster** [comment]
I kinda want to punt on this actually. Doesn't seem worth it...

**Comment by paulfalgout** [comment]
cool.  yeah.. on the fence myself

**Comment by paulfalgout** [comment]
With all of the nice refactors, I think this would clean some things up.  Time to reconsider?

**Comment by jasonLaster** [comment]
lets do it. will rebase

**Comment by jasonLaster** [comment]
@paulfalgout after thinking about it, I don't think this warrants tests or documentation as I'd like to keep the function internal and not exposed on `Marionette`. It'll be well tested as it will be used in lots of places.

**Comment by paulfalgout** [comment]
Sounds good.  Can the util filename be _'d and the getValue is now public

**Comment by jasonLaster** [comment]
sure

**Comment by paulfalgout** [comment]
oh sorry.. actually the API for both getOption and getValue is different in this context.  neither accept the target argument and should be called with context.. and I think getValue now accepts n args instead of an array

**Comment by jasonLaster** [comment]
sorry, tried and failed again i believe. looking at getOption again, it became clear that it expected a context `this`, which getOptionValue wouldn't have... Anyways, once again, not sure about the value here... any suggestions.

File: src/utils/_getOptionValue.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Composes getOption with the intent of
+// evaluating a target's property or option.
+// Useful when class property like `events` or `regions`
+// could either be an object or function.
+export default function(target, optionName, params) {
```
Comment (paulfalgout) [inline src/utils/_getOptionValue.js]: maybe
```js
export default function(target, optionName, ...args) {
  var options = getOption.call(target, optionName);
  return getValue.apply(target, [option].concat(args));
}
```

Comment (jasonLaster) [inline src/utils/_getOptionValue.js]: yea, honestly, not feeling it... I think `this.getValue(this.getOption('foo'))` is fine in most cases

Comment (paulfalgout) [inline src/utils/_getOptionValue.js]: :+1:

**Pull request closed without merging**